pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go

```diff
@@ -273,15 +273,14 @@ func RaftEventFromMsgStorageAppendAndMsgApps(
 	outboundMsgs []raftpb.Message,
 	logSnapshot RaftLogSnapshot,
 	msgAppScratch map[roachpb.ReplicaID][]raftpb.Message,
+	replicaStateInfoMap map[roachpb.ReplicaID]ReplicaStateInfo,
```

Comment:
Is it essential that these two fields are maps, or slices would suffice?

If maps are essential, do we need them to be separate maps, or something like `map[ReplicaID]AllAboutThisReplica` would suffice? Could apply the same "scratch" policy to both in one place.